components/service-operator/apis/cache/v1beta1/elasticache_cluster_types.go

```diff
@@ -0,0 +1,316 @@
+	"github.com/alphagov/gsp/components/service-operator/internal/env"
+	"github.com/alphagov/gsp/components/service-operator/internal/object"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&ElasticacheCluster{}, &ElasticacheClusterList{})
+}
+
+const (
+	ElasticacheClusterResourceName                    = "ElasticacheCluster"
+	CacheSubnetGroupParameterName                     = "CacheSubnetGroup"
+	VPCSecurityGroupIDParameterName                   = "VPCSecurityGroupID"
+	ElasticacheClusterRedisPrimaryHostnameOutputName  = "ClusterPrimaryRedisHostname"
+	ElasticacheClusterRedisPrimaryPortOutputName      = "ClusterPrimaryRedisPort"
+	ElasticacheClusterRedisReadHostnamesOutputName    = "ClusterReadRedisHostnames"
+	ElasticacheClusterRedisReadPortsOutputName        = "ClusterReadRedisPorts"
+	AuthTokenSecretResourceName                       = "AuthTokenSecret"
+	ElasticacheClusterRedisAuthTokenOutputName        = "SecretAuthToken"
+)
+
+// ensure implements required interfaces
+var _ cloudformation.Stack = &ElasticacheCluster{}
+var _ object.SecretNamer = &ElasticacheCluster{}
+var _ cloudformation.StackSecretOutputter = &ElasticacheCluster{}
+var _ cloudformation.ServiceEntryCreator = &ElasticacheCluster{}
+
+// AWS allows specifying configuration for the elasticache cluster
+type ElasticacheClusterAWSSpec struct {
+	// NodeType defines the amount of RAM and CPUs nodes in the cluster have as well as their network performance
+	NodeType string `json:"nodeType"`
+
+	// EngineVersion defines the version of Redis running in the cluster.
+	EngineVersion string `json:"engineVersion"`
+
+	// NumCacheClusters defines the number of clusters that belong to our replication group. A number between 2 and 6 inclusive.
+	NumCacheClusters int `json:"numCacheClusters"`
+
+	// PreferredMaintenanceWindow defines the weekly window during which maintenance is performed on the cluster. The minimum period is 60 minutes.
+	PreferredMaintenanceWindow string `json:"preferredMaintenanceWindow,omitempty"`
+}
+
+// ElasticacheClusterSpec defines the desired state of ElasticacheCluster
+type ElasticacheClusterSpec struct {
+	// AWS specific subsection of the resource.
+	AWS ElasticacheClusterAWSSpec `json:"aws,omitempty"`
+	// Secret name to be used for storing relevant instance secrets for further use.
+	Secret string `json:"secret,omitempty"`
+	// ServiceEntry name to be used for storing the egress firewall rule to allow tenant access to the cluster
+	ServiceEntry string `json:"serviceEntry,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// ElasticacheClusterList contains a list of ElasticacheCluster
+type ElasticacheClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ElasticacheCluster `json:"items"`
+}
+
+// +kubebuilder:object:root=true
+
+// ElasticacheCluster is the Schema for the ElasticacheCluster API
+type ElasticacheCluster struct {
```

Comment:
We should probably just call it `Redis` ... just like we call rds postgres `Postgres`